pkg/resources/machine/common.go

```diff
@@ -554,8 +556,24 @@ func GetKubevirtProviderConfig(_ *kubermaticv1.Cluster, nodeSpec apiv1.NodeSpec,
 	return config, nil
 }
 
-func getKubevirtProviderSpec(nodeSpec apiv1.NodeSpec, dc *kubermaticv1.Datacenter) (*runtime.RawExtension, error) {
-	config, err := GetKubevirtProviderConfig(nil, nodeSpec, dc)
+func extractKubeVirtOsImageURLOrDataVolumeNsName(namespace string, osImage string) string {
+	// config.VirtualMachine.Template.PrimaryDisk.OsImage.Value contains:
+	// - a URL
+	// - or a DataVolume name
+	// If config.VirtualMachine.Template.PrimaryDisk.OsImage.Value is a DataVolume, we need to add the namespace prefix
+	if _, err := url.ParseRequestURI(osImage); err == nil {
+		return osImage
+	}
+	// It's a DataVolume
+	// If it's already a ns/name keep it.
+	if nameSpaceAndName := strings.Split(osImage, "/"); len(nameSpaceAndName) >= 2 {
```

Comment:
nit: wouldn't it be safer to use namespace name in the split function? I'm thinking about a situation where `osImage` would contain `/` in its name which is probably not possible.
```suggestion
	if nameSpaceAndName := strings.Split(osImage, namespace); len(nameSpaceAndName) >= 2 {
```